File: framework/entity/Persistence.java
package framework.entity;

import java.util.*;

public abstract class Persistence {
  public Persistence() {
  }	
  public abstract void setDatabase(String database);

  public abstract Entity fetchEntity(Entity entity, String key) throws PersistenceException;

  public abstract void fetch(EntitySet set, Entity prototype, Condition condition) throws PersistenceException;
  
  public abstract void insert(Entity entity) throws PersistenceException;

  public abstract void update(Entity entity) throws PersistenceException;

  public abstract void delete(Entity entity) throws PersistenceException;

  public HashMap getValueTable(Entity entity) {
    return entity.getValueTable();
  }
}

File: src/Persistance/Persistence.cpp
#include "StdAfx.h"
#include "Persistence.h"


Persistence::Persistence() 
{
}

Persistence::~Persistence() 
{
}



  Entity* Persistence::fetchEntity(Entity* entity, CString key){return entity;}

//  void Persistence::Fetch(EntitySet set, Entity prototype, Condition condition){}
  void Persistence::Fetch(Entity* entity){}
    
  void Persistence::Insert(Entity* entity){}

  void Persistence::Update(Entity* entity){}
  void Persistence::Update(Entity* entity, CStringList* fieldOrder){}

  void Persistence::Delete(Entity* entity) {}

  void Persistence::Rename(Entity* entity){}
/*
  CMapCStringToOb* Persistence::getValueTable(Entity* entity) 
  {
    return entity->getValueTable();
  }
*/  

